pkg/controllers/vdb/dbaddnode_reconciler.go

```diff
@@ -113,30 +113,52 @@ func (d *DBAddNodeReconciler) findAddNodePods(scName string) ([]*PodFact, ctrl.R
 	return podList, ctrl.Result{}
 }
 
+// findExpectedNodeNames will return a list of pods that should have been in the database
+// before running db_add_node (which are also called expected nodes)
+func (d *DBAddNodeReconciler) findExpectedNodeNames() []string {
+	var expectedNodeNames []string
+
+	for _, v := range d.PFacts.Detail {
+		if v.dbExists {
+			expectedNodeNames = append(expectedNodeNames, v.name.Name)
+		}
+	}
+
+	return expectedNodeNames
+}
+
 // reconcileSubcluster will reconcile a single subcluster.  Add node will be
 // triggered if we determine that it hasn't been run.
 func (d *DBAddNodeReconciler) reconcileSubcluster(ctx context.Context, sc *vapi.Subcluster) (ctrl.Result, error) {
 	addNodePods, res := d.findAddNodePods(sc.Name)
 	if verrors.IsReconcileAborted(res, nil) {
 		return res, nil
 	}
+
+	expectedNodeNames := d.findExpectedNodeNames()
```

Comment:
Do we need to build the list here? Could we just call `d.findExpectedNodeNames` in `runAddNodeForPod`? It would save having to pass this down.